packages/blade/src/components/Dropdown/Dropdown.tsx

```diff
@@ -99,24 +98,22 @@ const _Dropdown = ({
 
   const dropdownBaseId = useId('dropdown');
 
-  useDidUpdate(() => {
-    onOpenChange?.(isOpen);
-    if (!isOpen) {
```

Comment:
With this logic, we were calling `onOpenChange` as well as we were actually closing and opening dropdown.

Ideally we should only call `onOpenChange` wherever state needs to change and then consumers add 

```jsx
<Dropdown isOpen={isDropdownOpen} onOpenChange={(isOpen) => { setIsDropdownOpen(isOpen) }} />
```

if someone doesn't call state change in onOpenChange, then it shouldn't close the dropdown.